### PR TITLE
Key cert parsing

### PR DESCRIPTION
**Pull request by grep-ir**
Right now this only supports output as PEM encoded data.  DER probably wouldn't be very template friendly anyways.  TODO for supporting a "chain" transformation to return the issuing chain.

### CODE REVIEW

File: templaterenderer/templaterenderer.go [hunk 1]
```diff
@@ -63,3 +97,134 @@ func RenderInline(templateContents string, resourceMap resource.ResourceMap) str
 
 	return result
 }
+
+func pemPrivateKeyFromPkcs12(b64pkcs12 string) string {
+	log.Print("Getting private key from PKCS12 data")
```
Comment (chrisjohnson) [inline templaterenderer/templaterenderer.go]: I've started to remove all the unnecessary noise from the logs. I was thinking we should sit down and think about what all we want to show up in the logs specifically during normal (non-error) operation. For now I don't think any of these methods need to be logging

File: templaterenderer/templaterenderer.go [hunk 2]
```diff
@@ -63,3 +97,134 @@ func RenderInline(templateContents string, resourceMap resource.ResourceMap) str
...
+
+	// The chain is the rest of the certs
+	var chain bytes.Buffer
+	for _, issuerBytes := range certAndKey.Certificate[1:] {
+		if err := pem.Encode(&chain, &pem.Block{Type: "CERTIFICATE", Bytes: issuerBytes}); err != nil {
+			log.Panicf("Failed to write data: %s", err)
+		}
+	}
+
+	//TODO make sure the chain is in the right order - each cert certifies the one preceding it
```
Comment (chrisjohnson) [inline templaterenderer/templaterenderer.go]: Is this something that needs to be addressed before merging this PR or is it just a nice to have?

Comment (grep-ir) [reply templaterenderer/templaterenderer.go]: For the chain to be useful we will need to implement this sorting.

File: templaterenderer/templaterenderer.go [hunk 3]
```diff
@@ -25,24 +24,55 @@ func RenderInline(templateContents string, resourceMap resource.ResourceMap) str
 	helpers := template.FuncMap{
 		"privateKey": func(name string) interface{} {
```
Comment (chrisjohnson) [inline templaterenderer/templaterenderer.go]: I put interface{} here as a stub but I think all three of these methods can have `string` as their return type

**Pull request merged**